app/lib/attachment_batch.rb

```diff
@@ -95,13 +96,25 @@ def remove_files
     # objects can be processed at once, so we have to potentially
     # separate them into multiple calls.
 
+    retries = 0
     keys.each_slice(LIMIT) do |keys_slice|
-      logger.debug { "Deleting #{keys_slice.size} objects" }
-
-      bucket.delete_objects(delete: {
-        objects: keys_slice.map { |key| { key: key } },
-        quiet: true,
-      })
+      begin
+        logger.debug { "Deleting #{keys_slice.size} objects" }
+        bucket.delete_objects(delete: {
+          objects: keys_slice.map { |key| { key: key } },
+          quiet: true,
+        })
+      rescue => e
+        retries += 1
+        if retries < MAX_RETRY
+          logger.debug "Retry #{retries}/#{MAX_RETRY} after #{e.message}"
+          sleep 2**retries
+          retry
+        else
+          logger.error "Batch deletion from S3 failed after #{e.message}"
+          raise e
+        end
+      end
```

Comment:
No need to add another nesting level:
```suggestion
      logger.debug { "Deleting #{keys_slice.size} objects" }

      bucket.delete_objects(delete: {
        objects: keys_slice.map { |key| { key: key } },
        quiet: true,
      })
    rescue => e
      retries += 1
 
      if retries < MAX_RETRY
        logger.debug "Retry #{retries}/#{MAX_RETRY} after #{e.message}"
        sleep 2**retries
        retry
      else
        logger.error "Batch deletion from S3 failed after #{e.message}"
        raise e
      end
```